hedera-node/src/main/java/com/hedera/services/state/logic/RecordStreaming.java

```diff
@@ -75,9 +99,43 @@ public void run() {
 	}
 
 	public void stream(final RecordStreamObject rso) {
+		if (isInNewBlock(rso)) {
+			Hash computedHash = new Hash();
+			try {
+				computedHash = curRunningHashLeaf.getRunningHash().getFutureHash().get();
+			} catch (final InterruptedException e) {
+				log.error("Error in computing hash for block #{}", curNetworkCtx.getBlockNo());
+				Thread.currentThread().interrupt();
+				throw new RuntimeException(String.format("Failed to calculate hash for block #%d", curNetworkCtx.getBlockNo()));
+			}
+
+
+			curNetworkCtx.cachePreviousBlockHash(computedHash);
+			curNetworkCtx.incrementBlockNo();
+			curNetworkCtx.setFirstConsTimeOfCurrentBlock(rso.getTimestamp());
+			log.info("Beginning block #{} @ {}", curNetworkCtx.getBlockNo(), rso.getTimestamp());
+		}
+
 		runningHashUpdate.accept(rso.getRunningHash());
+		lastConsensusTimestamp = rso.getTimestamp();
```

Comment:
This `lastConsensusTimestamp` also needs to be in state, just like the `firstConsTimeOfCurrentBlock` field; and it might be better named as `latestConsTimeOfCurrentBlock`.

(I.e., when a node reconnects, its `(firstConsTime, latestConsTime)` block metadata must be in-sync with nodes that never disconnected; an instance variable of `RecordStreaming` would not have this property.)